src/docdb/utils/managedIdentityUtils.ts

```diff
@@ -0,0 +1,113 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+// eslint-disable-next-line import/no-internal-modules
+import { AzureCliCredential, ManagedIdentityCredential } from '@azure/identity';
+import * as vscode from 'vscode';
+import { ext } from '../../extensionVariables';
+import { AuthenticationMethod, type CosmosDBManagedIdentityCredential } from '../getCosmosClient';
+
+// Module-level variable to cache the result of running on Azure check
+let isRunningOnAzure: boolean | undefined = undefined;
+
+export async function getIsRunningOnAzure(): Promise<boolean> {
+    // Return cached result if available
+    if (isRunningOnAzure !== undefined) {
+        return isRunningOnAzure;
+    }
+    isRunningOnAzure = false;
+
+    try {
+        // Request with timeout
+        const controller = new AbortController();
+        const timeoutId = setTimeout(() => controller.abort(), 2000);
+
+        const response = await fetch('http://169.254.169.254/metadata/versions', {
```

Comment:
Is it for testing purpose?

Reply:
No it's an actual service which is only available in Azure VMs, see https://learn.microsoft.com/en-us/azure/virtual-machines/instance-metadata-service, I'll add a comment.